packages/plugin-core/src/WorkspaceWatcher.ts

```diff
@@ -312,16 +317,15 @@ export class WorkspaceWatcher {
     }
 
     const content = event.document.getText();
-    const matchFM = NoteUtils.RE_FM;
-    const matchOuter = content.match(matchFM);
-    if (!matchOuter) {
-      return { changes: [] };
-    }
     const match = NoteUtils.RE_FM_UPDATED.exec(content);
     let changes: TextEdit[] = [];
 
-    // update the `updated` time in frontmatter
-    if (match && parseInt(match[1], 10) !== note.updated) {
+    // update the `updated` time in frontmatter if it exists and content has changed
+    if (
+      this._textDocumentService.containsFrontmatter(event.document) &&
```

Comment:
Should this check then happen before we run the regexp (and even get document contents)?

Reply:
Good point, makes sense to me